File: AULA_03/Haskell/NotaConceito.hs
conceito :: Bool -> Bool -> String
conceito _ True = "O"
conceito True False = "A"
conceito False False  = "R"

pegaAluno x n limiar maxFaltas  
       | x==n = do putStrLn ""
       | x<n = do
               putStrLn "Entre com a nota:  "
               line <- getLine
               let nota  = read line
               putStrLn "Entre com as faltas: "
               line <- getLine
               let faltas  = read line
              
               let c = conceito ((nota :: Float) > (limiar :: Float))  ((faltas :: Int) > (maxFaltas :: Int))
               putStrLn $ "Aluno " ++ (show x) ++ " conceito: " ++ c
               pegaAluno (x+1) n limiar maxFaltas

main = do
       putStrLn "Entre com limiar: "
       line <- getLine
       let limiar  = read line
       putStrLn "Entre com max. faltas: "
       line <- getLine
       let maxFaltas  = read line
       putStrLn "Numero de alunos: "
       line <- getLine
       let alunos = read line
       pegaAluno 0 alunos limiar maxFaltas

File: AULA_03/C/NotaConceito.c
#include <stdio.h>

int main(){

  float nota, limiar;
  int faltas, maxFaltas;
  int nAlunos, i;

  printf("Digite o limiar de notas: ");
  scanf("%f",&limiar);
  printf("Digite o limiar de faltas: ");
  scanf("%d",&maxFaltas);
  printf("Quantos alunos? ");
  scanf("%d",&nAlunos);

  for( i=1;i<=nAlunos;i++ ){
    printf("Digite a nota: ");
    scanf("%f",&nota);
    printf("Digite as faltas:");
    scanf("%d",&faltas);

    printf("Aluno %d conceito ",i);

    if( faltas > maxFaltas ){
      printf("O\n");
    }else{
      if( nota > limiar ){
        printf("A\n");
      }else{
        printf("R\n");
      }
    }
  }

  return 0;
}
